### PR TITLE
Fix ReadTheDocs PDF build

### LINKED ISSUE #3: Setup Read the Docs
The `main` branch should automatically be deployed to Read the Docs.

This issue tracks this effort.

### LINKED ISSUE #58: docs: PDF documentation broken
I can't seem to be able to download the PDF documentation generated by read-the-docs. The browser doesn't open the file, and when I try to download it directly, my PDF reader says it's corrupted.

---
*Reported from: https://canonical-ubuntu-packaging-guide.readthedocs-hosted.com/en/latest/*

### PR DESCRIPTION
**Pull request by SecondSkoll**
Fixes issue outlined in #58

### CODE REVIEW

**Comment by SecondSkoll** [comment]
RTD is being a real pain. So, the issue was bound to RTD using the root of the git clone and then generating files in `_readthedocs/<format>`. Sadly this can't be changed - and while for HTML this doesn't have any negative effects - it does for the PDF generation which, due to how the title page is implemented, uses relative links to some files.

~~So this fix changes the relative links to reference the current directory, by pre-copying the relevant files into the directory used by RTD (because RTD for some reason isn't registering the existence of files included with `latex_additional_files` so the images and fonts just don't exist for the RTD PDF build). It also implements some changes to support local generation.~~

~~The next issue is that while this does work, it doesn't work on a PR as RTD does not build a PDF on a PR - and because PDF generation would automatically clean up that output folder it gets very upset at the fact we put files in there for them to use to generate the PDF. This is why there's a bunch of commits here - as I didn't discover this part until I submitted the PR... So, if you can't squash this I can submit a new, clean PR instead.~~

~~Weirdly, RTD offers no flexibility in how it handles cases where it does or doesn't generate PDFs, so there's a conditional cleanup in `post_build`.~~

~~All this because of what I assume is a bug in how RTD handles `latex_additional_files` (aka it doesn't handle them at all, and ignores them completely).~~

~~TL;DR: This is a hacky fix on top of a hacky fix all due to some issues in RTD. I'll see if I can get this fixed and circle back to this project and implement fixes properly - but it might take some time. It will all work for now.~~

I really thought RTD was just ignoring some files listed in `latex_additional_files`, but it turns out I need an updated prescription for my glasses... Well, this should all work which is the important thing. I've verified manual output and RTD build using a [fork of the starter pack](https://github.com/SecondSkoll/sphinx-docs-starter-pack/tree/rtd-subdir) where I have RTD builds on. You can verify the built PDF by looking at the RTD versions in the bottom left menu [here](https://canonical-test-starter-pack-pdfgen.readthedocs-hosted.com/en/rtd-subdir/).

File: docs/Makefile
Comment (dviererbe) [inline docs/Makefile]: nitpick: `. $(VENV)` is not needed for `rm`, `find`, `mv` operations and `@` swallows logging the commands executed

```suggestion
	. $(VENV); sphinx-build -M latexpdf "$(SOURCEDIR)" "_build" $(SPHINXOPTS)
	rm ./_build/latex/front-page-light.pdf || true
	rm ./_build/latex/normal-page-footer.pdf || true
	find ./_build/latex -name "*.pdf" -exec mv -t ./ {} +
	rm -r _build
```

File: .readthedocs.yaml
Comment (dviererbe) [inline .readthedocs.yaml]: question: Can this be dropped?

**Review by dviererbe — CHANGES_REQUESTED** [review]
LGTM, there are two nitpicks I would like to be addressed before merging.

note: I can not access https://canonical-test-starter-pack-pdfgen.readthedocs-hosted.com/en/rtd-subdir/ and the PR preview does not show a download as PDF link, but I trust you that it works now.

note: The link check errors are all expected and should not block the merge. They are fixed in #53 (not merged yet).

File: .readthedocs.yaml
Comment (SecondSkoll) [reply .readthedocs.yaml]: Yes, sorry I didn't notice the whitespace.

Comment (SecondSkoll) [inline .readthedocs.yaml]:
```suggestion

```
Fixes whitespace

**Review by SecondSkoll — COMMENTED** [review]
You can try https://canonical-test-starter-pack-pdfgen.readthedocs-hosted.com/_/sharing/G3ZK1L59yDc9fFutCcA3kuZ5Lr7C8J1L?next=/en/rtd-subdir/ to check the build preview - I'm not sure what level of access the temporary sharing URL provides though.

**Review by dviererbe — APPROVED** [review]
> You can try https://canonical-test-starter-pack-pdfgen.readthedocs-hosted.com/_/sharing/G3ZK1L59yDc9fFutCcA3kuZ5Lr7C8J1L?next=/en/rtd-subdir/ to check the build preview - I'm not sure what level of access the temporary sharing URL provides though.

This works for me now and I could verify that the download PDF link works :)

Merging now, thanks again for fixing this!

**Pull request merged**
